web/src/main/ReplyBox.js

```diff
@@ -299,6 +311,173 @@ class ReplyBox extends React.Component {
+            username={reply.author}
+            avatar={reply.avatar}
+            size={isChild ? "middle" : ""}
+          />
+        </div>
+        <div style={{ marginLeft: isChild ? "48px" : "60px" }}>
+          <div className="fr">
+            {this.props.account !== null &&
+            this.props.account !== undefined &&
+            this.props.account.id !== reply?.author ? (
+              reply?.thanksStatus === false ? (
+                <div
+                  id={`thank_area__${reply.id}`}
+                  className="thank_area"
+                  style={{ marginRight: "10px" }}
+                >
+                  <a
+                    href="#;"
+                    className="thank"
+                    style={{
+                      color: "#ccc",
+                      display: Setting.PcBrowser ? "" : "none",
+                      marginRight: "10px",
+                    }}
+                  >
+                    {i18next.t("reply:ignore")}
+                  </a>
+                  <a
+                    href="#;"
+                    onClick={() => this.thanksReply(reply.id, reply.author)}
+                    className="thank"
+                  >
+                    {Setting.PcBrowser ? (
+                      i18next.t("reply:thank")
+                    ) : (
+                      <img
+                        src={Setting.getStatic("/img/heart_neue.png")}
+                        width="16"
+                        style={{ verticalAlign: "bottom" }}
+                        alt={i18next.t("reply:thank")}
+                      />
+                    )}
+                  </a>
+                </div>
+              ) : (
+                <div
+                  id={`thank_area__${reply.id}`}
+                  className="thank_area thanked"
+                >
+                  {i18next.t("reply:thanked")}
+                </div>
+              )
+            ) : null}
+            {reply?.deletable ? (
+              <div
+                id={`thank_area__${reply.id}`}
+                className="thank_area"
+                style={{ marginRight: "10px" }}
+              >
+                <a
+                  href="#;"
+                  className="delete"
+                  style={{ marginRight: "10px" }}
+                  onClick={() => this.deleteReply(reply.id)}
+                >
+                  {i18next.t("reply:Delete")}
+                </a>
+                <a href={`/edit/reply/${reply.id}`} className="edit">
+                  {i18next.t("reply:Edit")}
+                </a>
+              </div>
+            ) : null}
+            {this.props.account !== undefined && this.props.account !== null ? (
+              <a
+                href="#;"
+                onClick={() => {
+                  this.handleClick(`@${reply.author} `);
+                  this.setState({
+                    parent: {
+                      id:
+                        depth < Conf.ReplyMaxDepth - 1
```

Comment:
The parent id shouldn't be affected by the `Conf` Instead, you could select the different style depending on the `depth` instead.
Our database should just record the truth.

Reply:
fixed